addon-test-support/index.js

```diff
@@ -0,0 +1,37 @@
+import { getContext } from '@ember/test-helpers';
+import { _authenticateSession, _currentSession, _invalidateSession } from './-private';
+
+/**
+ * Authenticates the session.
+ *
+ * @param {Object} sessionData Optional argument used to mock an authenticator
+ * response (e.g. a token or user).
+ * @return {Promise}
+ * @public
+ */
+export function authenticateSession(sessionData) {
```

Comment:
This looks good but when I suggested to extract the logic to some functions that receive an `owner` is because I thought we could also reuse those functions in the existing helpers, but it seems we can't because for backwards compatibility reasons the old helpers have to use `app.testHelpers.wait`.

In that case, I think that extracting the logic into separate functions, while correct, it's not necessary.

Other than that, this looks perfect.

Reply:
@cibernox @marcoow great feedback. implemented feedback, squashed commits 👍  in https://github.com/simplabs/ember-simple-auth/pull/1519/commits/ce6893a799d62d7e56f6682330bd6567c67c5293